cpp/ql/test/experimental/query-tests/Security/CWE/CWE-200/semmle/tests/test2.cpp

```diff
@@ -0,0 +1,18 @@
+typedef int FILE;
+FILE *fopen(const char *filename, const char *mode);
+int umask(int pmode);
+int chmod(char * filename,int pmode);
+int fprintf(FILE *fp,const char *fmt, ...);
+char *fgets(char *str, int num, FILE *stream);
+int fclose(FILE *stream);
+
+int main(int argc, char *argv[])
+{
+  FILE *fp;
+  char buf[128];
+  fp = fopen("myFile.txt","w"); // BAD [NOT DETECTED]
+  fgets(buf,128,fp); 
+  fprintf(fp,"%s\n","data to file");
```

Comment:
Is this supposed to be a test of reading from or writing to the file?

Reply:
I think that the presence of a function for reading should exclude the detection.
I think such a test is more complete, but I can remove the recording function.